File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/card/UpdateCardUseCase.java
package com.lattis.ellipse.domain.interactor.card;

import com.lattis.ellipse.data.network.model.response.BasicResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.CardRepository;

import javax.inject.Inject;

import io.reactivex.Observable;


public class UpdateCardUseCase extends UseCase<BasicResponse> {

    private CardRepository cardRepository;


    public UpdateCardUseCase setCardId(int cardId) {
        this.cardId = cardId;
        return this;
    }

    private int cardId;


    @Inject
    public UpdateCardUseCase(ThreadExecutor threadExecutor,
                             PostExecutionThread postExecutionThread,
                             CardRepository cardRepository) {
        super(threadExecutor, postExecutionThread);
        this.cardRepository = cardRepository;
    }


    @Override
    protected Observable<BasicResponse> buildUseCaseObservable() {
        return cardRepository.updateCard(cardId);
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/card/UpdateCardUseCase.kt
package com.lattis.domain.usecase.card

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.CardRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class UpdateCardUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val cardRepository: CardRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var cardId = 0
    fun setCardId(cardId: Int): UpdateCardUseCase {
        this.cardId = cardId
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return cardRepository.updateCard(cardId)
    }

}